services/utils.py

```diff
@@ -101,7 +102,9 @@ def get_mirror(status, id, row):
     else:
         host = user_media_url('addons')
 
-    return posixpath.join(host, str(id), row['filename'])
+    url = posixpath.join(host, str(id), row['filename'])
+    params = urllib.urlencode({'filehash': row['hash']})
+    return '{0}?{1}'.format(url, params)
```

Comment:
I have absolutely nothing against that, but what would be the benefit? It's not a public facing url, it's the one that's answered to the client (Firefox) on automatic version update checks.